src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/SharedEntrypoint.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.Extension.Common
+{
+    internal sealed class SharedEntrypoint<T> where T : class, IEgressProviderOptions, new()
+    {
+        private static Stream StdInStream;
+        private static CancellationTokenSource CancelSource = new CancellationTokenSource();
+
+        private static EgressProvider _provider;
+        private static Action<ExtensionEgressPayload, T> _configureOptions;
+
+        public static async Task<int> Entrypoint(string[] args, EgressProvider provider, Action<ExtensionEgressPayload, T> configureOptions)
```

Comment:
I think this whole method should be moved back to the Program classes for each extension. This particular class should focus on egressing whereas the extension can focus on translating command line to invoking this class (separation of concerns). Maybe keep the egress command creation part in this class; follow the pattern we use for the unit test app: https://github.com/dotnet/dotnet-monitor/blob/main/src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/AsyncWaitScenario.cs

I would rename this class to something like `EgressHelper`.

The `Egress` method in this class can be updated to be `private static async Task<int> Egress<TOptions>(EgressProvider provider, Action<ExtensionEgressPayload, TOptions> configureOptions = null)`

Reply:
This should be resolved, but I wasn't entirely sure how you wanted the command work split out. Let me know if you had something different in mind and I can make some more adjustments.